### DISCUSSION TITLE
keywords matching irrespective of quotes

### DISCUSSION THREAD

**Discussion by techpranav** [antlr/antlr4#4190] [Q&A]
Hi, 

Like caseInsensitive, is there any way to match the keywords irrespective of surrounded by quotes.
for ex.

select , "select" , 'select'

**Comment by KvanTTT** [comment]
No, because it's very custom logic, but you can use the following definition:

```g4
SELECT: 'SELECT' | '"SELECT"' | '\'SELECT\'';
```

**Reply by techpranav** [reply]
Yes, currently I am using this as a workaround.
But for all the keywords, do I need to specify like this?

**Reply by KvanTTT** [reply]
Unfortunately yes.

**Comment by jimidle** [comment]
It’s not even unfortunate Ivan. This is a very specific requirement. Being
able to do this in some automated way makes no sense.

What language are we parsing here? If it’s a self designed language, then
just stop doing that. It is a really, er, “problematic” design.

**Comment by jimidle** [comment]
It’s not a work round. That’s what you need to do if the language you have
to deal with has required this. It is a very stupid thing to do, but I’ve
seen worse.
